crates/bevy_mesh/src/vertex.rs

```diff
@@ -138,6 +138,25 @@ pub(crate) struct MeshAttributeData {
     pub(crate) values: VertexAttributeValues,
 }
 
+/// Compute a vector whose direction is the normal of the triangle formed by
+/// points a, b, c, and whose magnitude is double the area of the triangle. This
+/// is useful for computing smooth normals where the contributing normals are
+/// proportionate to the areas of the triangles as [discussed
+/// here](https://iquilezles.org/articles/normals/).
+///
+/// Question: Why double the area? Because the area of a triangle _A_ is
+/// determined by this equation:
+///
+/// _A = |(b - a) x (c - a)| / 2_
+///
+/// By computing _2 A_ we avoid a division operation, and when calculating the
+/// the sum of these vectors which are then normalized, a constant multiple has
+/// no effect.
+pub(crate) fn face_area_normal(a: [f32; 3], b: [f32; 3], c: [f32; 3]) -> [f32; 3] {
+    let (a, b, c) = (Vec3::from(a), Vec3::from(b), Vec3::from(c));
+    (b - a).cross(c - a).into()
+}
+
 pub(crate) fn face_normal(a: [f32; 3], b: [f32; 3], c: [f32; 3]) -> [f32; 3] {
```

Comment:
You could probably reuse face_area_normal here and just call .normalize() on the output.

Reply:
In the body of `face_normal()`? I thought about that, but I figured it'd be an extra function call. Maybe that's C thinking though. However, I'd have to call `Vec3::from()` again to call normalize from `face_area_normal()` since we'd outputting `[f32; 3]`.

Happy to do it if you think it's worth it.

Reply:
Oh, right, I missed that it outputs a `[f32:3]`. Although, I wouldn't be surprised if the compiler just optimizes all of it out. Same thing for the function call. Could add an `inline` annotation if it's an issue. Doesn't really matter though, it's just a few lines so duplication is fine.

Reply:
I'll add the inlines and make them pub.